graph-kiota/kiota.md

```diff
@@ -0,0 +1,180 @@
+        -  Middlewares: AuthHandler, Retry, Redirect, Compression, Caching, Telemetry
+    
+```
+class HttpCore implements kiota.abstractions.HttpCore {
+    constructor(authProvider, parseNodeFactory, serializationWriteFactory, httpClient);
+
+    sendAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType>;
+    sendCollectionAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType[]>;
+    sendPrimitiveAsync<ResponseType>(requestInfo: RequestInformation, responseType: "string" | "number" | "boolean" | "Date" | "ReadableStream", responseHandler: ResponseHandler | undefined): Promise<ResponseType>;
+    sendNoResponseContentAsync(requestInfo: RequestInformation, responseHandler: ResponseHandler | undefined): Promise<void>;
+    enableBackingStore(backingStoreFactory?: BackingStoreFactory | undefined): void;
+}
+
+class HttpClient{
+    middlewares : Middlewares();
+    send() //calls the MiddlewareChain to make a  request with the native HttpClient;
+}
+```
+
+Kiota Generated API Client (This will be the GraphServiceClient)
+
+```
+class APIClient {
+
+    httpCoreInstance : HttpCore
+
+    SampleRequestBuilder(httpCoreInstance);
+    ExampleRequestBuilder(httpCoreInstance);
+}
+```
+
+- Kiota serializations 
+    - This library contains implementation of the kiota abstractions serialization interfaces
+    - Currently targets `JSON` format.
+    - Other formats to consider:
+        - Multipart payload
+        - Base64 encoded string
+        - text/plain
+        - xml
+
+- Kiota auth providers
+
+### Example of how we use the Kiota libraries:
+
+1. Clone the Kiota repo, build the project and navigate the path containing the `kiota.exe`
+2. Generate the request builder using the command:
+```
+kiota.exe -d ../msgraph-sdk-powershell/openApiDocs/v1.0/mail.yml --language csharp -o ../somepath -n namespaceprefix
+```
+3. The generation should contain the `APIClient` class and request builders.
+
+5. Create a demo class to test the `APIClient`.
+
+4. Import the `KiotaCore` library
+- [dotnet](https://github.com/microsoft/kiota/packages/826853)
+- [TypeScript](https://github.com/microsoft/kiota/packages/827040)
+
+Example:
+```
+import { HttpCore } from '@microsoft/kiota-http-fetch';
+```
+
+5. Import the serialization library
+- [dotnet](https://github.com/microsoft/kiota/packages/826855)
+- [TypeScript](https://github.com/microsoft/kiota/packages/827041)
+
+```
+import {JsonParseNodeFactory, JsonSerializationWriterFactory} from '@microsoft/kiota-serialization-json';
+```
+
+Example:
+
+6. Import the AuthProvider library or implement the IAuthProvider interface to create a custom AuthProvider class.
+
+Find a suitable Kiota Authentication library in https://github.com/microsoft/kiota/packages
```

Comment:
```suggestion
Find a suitable Kiota Authentication library in https://github.com/microsoft/kiota/tree/main/authentication/typescript/azure
```

We could add a rollup of all the authentication packages in the /authentication folder, and one by language in the /authentication/language folder if it makes things easier.